src/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -1180,6 +1182,40 @@ protected SAMBinaryTagAndValue getBinaryAttributes() {
     }
 
     /**
+     * Interval covering the aligned, unclipped, bases in this read as reported by getAlignmentStart() and getAlignmentEnd()
+     * @return the aligned interval or null if getReadUnmappedFlag()
+     */
+    public SimpleInterval getLocation() {
+        if (getReadUnmappedFlag()) {
+            return null;
+        } else {
+            return new SimpleInterval(mReferenceName, mAlignmentStart, mAlignmentEnd);
+        }
+    }
+
+    /**
+     * @return get
+     */
+    @Override
+    public String getContig() {
+        if( getReadUnmappedFlag()) {
+            return null;
+        } else {
+            return getReferenceName();
```

Comment:
This seems fine given the disclaimer in the comments for this class that `Only if the readUnmappedFlag is false can the reference name/index and alignment start be interpreted as indicating an actual alignment position.`